paramiko/config.py

```diff
@@ -80,6 +81,26 @@ def parse(self, file_obj):
                 # at the end (for compatibility with issue #415). After 3.x, it
                 # will simply not get stripped, leaving a nice explicit marker.
                 host['config'][key] = None
+            elif key == 'include':
+                # support for Include directive in ssh_config
+                path = value
+                # according to SSH documentation if the path is relative, we look forward in ~/.ssh
+                if os.path.isabs(path) == False and path[0] != '~':
+                    path = '~/.ssh/'+path
+                #expand the user home path
+                path = os.path.expanduser(path)
+                if parsed_files == None:
+                    parsed_files = []
+
+                #parse every included file
+                for filename in glob.iglob(path):
+                    if os.path.isfile(filename):
+                        if filename in parsed_files:
+                            raise Exception("Include loop detected in ssh config file: %s" % filename)
```

Comment:
This should be some more specific exception class, using just `Exception` is almost never useful (it's impossible to catch without catching everything else too!); not sure what works best though. Maybe ValueError, or a new custom exception class (meh).